site/app/controllers/admin/AdminGradeableController.php

```diff
@@ -223,6 +223,7 @@ private function editPage(Gradeable $gradeable, $semester, $course, $nav_tab = 0
         $this->core->getOutput()->addInternalJs('admin-gradeable-updates.js');
         $this->core->getOutput()->addInternalCss('admin-gradeable.css');
         $this->core->getOutput()->renderTwigOutput('admin/admin_gradeable/AdminGradeableBase.twig', [
+            'core' => $this->core,
```

Comment:
Is there a reason why 'core' is getting passed to this twig file? 
Core typically contains a lot of information and we want to try restricting its usage to controller classes and propagate only relevant information to views so we don't risk exposing more information to the user than intended

Reply:
I wanted to use the `buildCourseUrl()` function in the twig file.  I don't agree that we shouldn't pass core to the twig file.  I've seen many cases where the view is passing 20 parameters (all information collected from core) to the twig where it could just be passing the core a single time.

Reply:
What out of core do you want or need in twig files? Are the 20 parameters just different urls? I agree we should figure out how to slim down some of the huge parameter lists, but do not want to pass in core unless there's a tremendously good reason as I have a strong feeling it's going to lead to abuses of things like running queries in the templates like we've got queries inside the views.

Reply:
I propose we create a white list of functions that would be useful inside the twig file, for example `buildCourseUrl()`, and maybe keep them in a `json` or `const` multi-dimensional array or something so they stay organized.  Then `Output::loadTwig()` can loop over the functions and register them as Twig functions the same way a few other functions are already being registered in there.

Reply:
Yes, but before bothering with that, I would appreciate if you could take a stab at answering:

> What out of core do you want or need in twig files? Are the 20 parameters just different urls?

Off the top of my head, I could see providing a twig function for `buildUrl` and `buildCourseUrl` and then a twig variable for csrf token. I am loathe to allow `getUser` or `getConfig` as I firmly believe we should pass models in to templates to minimize coupling.

Reply:
At this time I can only think of `buildUrl` and `buildCourseUrl`.  Not sure about the csrf token as I think its getting loaded in somewhere.  It seems to always been available inside the browser even when I haven't explicitly defined it.

Reply:
I removed core from the twig so we can move ahead with this PR.